### PR TITLE
Allow to have an empty path when convert a CONNECT request

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

CONNECT requests have no path defined as stated in the HTTP/2 spec, at the moment we will throw an exception if we try to convert such a request to HTTP/1.1

Modifications:

- Don't throw an exception if we try to convert a HTTP/2 CONNECT request that has no path
- Add unit test

Result:

Related to https://github.com/netty/netty-incubator-codec-http3/pull/112.

### CODE REVIEW

**Review by chrisvest — APPROVED** [review]

**Review by hyperxpro — APPROVED** [review]

File: codec-http2/src/main/java/io/netty/handler/codec/http2/HttpConversionUtil.java [hunk 1]
```diff
@@ -268,6 +268,15 @@ public static FullHttpRequest toFullHttpRequest(int streamId, Http2Headers http2
         return toFullHttpRequest(streamId, http2Headers, alloc.buffer(), validateHttpHeaders);
     }
 
+    private static String extractPath(CharSequence method, Http2Headers headers) {
+        if (HttpMethod.CONNECT.asciiName().contentEqualsIgnoreCase(method)) {
+            return "/";
```
Comment (ejona86) [inline codec-http2/src/main/java/io/netty/handler/codec/http2/HttpConversionUtil.java:273]: Why doesn't this return :authority? That is what is expected in HTTP/1.1.

Comment (normanmaurer) [reply codec-http2/src/main/java/io/netty/handler/codec/http2/HttpConversionUtil.java:273]: @ejona86 not sure I understand... can you show the related rfc ?

Comment (ejona86) [reply codec-http2/src/main/java/io/netty/handler/codec/http2/HttpConversionUtil.java:273]: https://tools.ietf.org/html/rfc7231#section-4.3.6

```
   A client sending a CONNECT request MUST send the authority form of
   request-target (Section 5.3 of [RFC7230]); i.e., the request-target
   consists of only the host name and port number of the tunnel
   destination, separated by a colon.  For example,

     CONNECT server.example.com:80 HTTP/1.1
     Host: server.example.com:80
```

Note that HTTP/1.1 doesn't use the Host header for performing the actual socket connect().

Comment (ejona86) [reply codec-http2/src/main/java/io/netty/handler/codec/http2/HttpConversionUtil.java:273]: I guess I should also include [the part from HTTP/2](https://tools.ietf.org/html/rfc7540#section-8.3):

```
   o  The ":authority" pseudo-header field contains the host and port to
      connect to (equivalent to the authority-form of the request-target
      of CONNECT requests (see [RFC7230], Section 5.3)).
```

Comment (normanmaurer) [reply codec-http2/src/main/java/io/netty/handler/codec/http2/HttpConversionUtil.java:273]: @ejona86 I guess what you say here is that for connect we should set the `Host:` header to the value of `:authority` correct ? The path would still be just `/` as its used for the uri, tho...

Comment (ejona86) [reply codec-http2/src/main/java/io/netty/handler/codec/http2/HttpConversionUtil.java:273]: I was most concerned about the `CONNECT server.example.com:80 HTTP/1.1` line. The uri of DefaultHttpRequest seems it should be copied from :authority. That isn't a "path" but the argument to DefaultHttpRequest is uri, not path, so I don't see any need to focus on it being a path (although I'm seriously not very familiar with DefaultFullHttpRequest).

I'd expect Host header to already be handled, but that is just an assumption.

Comment (normanmaurer) [reply codec-http2/src/main/java/io/netty/handler/codec/http2/HttpConversionUtil.java:273]: @ejona86 ah I miss-understood you... This makes sense. Let me fix this

Comment (normanmaurer) [reply codec-http2/src/main/java/io/netty/handler/codec/http2/HttpConversionUtil.java]: @ejona86 PTAL again

**Review by ejona86 — APPROVED** [review]

**Pull request merged**